app/models/llms/basic_workflow/integration.py

```diff
@@ -0,0 +1,391 @@
+        
+        Args:
+            user_input: Raw user input string
+            
+        Returns:
+            str: Generated itinerary text
+        """
+        with timer("enhanced itinerary generation"):
+            preferences = self.parse_travel_request(user_input)
+            
+            detailed_prompt = self._create_user_prompt(
+                destination=preferences['destination'],
+                duration=preferences['duration'],
+                interests=preferences['interests'],
+                budget=preferences['budget'],
+                travel_style=preferences['travel_style'],
+                additional_context=preferences['additional_context']
+            )
+            
+            return self.generate_itinerary(detailed_prompt)
+
+
+def save_to_session_history(session_history: list, user_input: str, itinerary: str, preferences: Dict[str, str]):
+    session_entry = {
+        'timestamp': datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+        'user_input': user_input,
+        'preferences': preferences,
+        'itinerary': itinerary
+    }
+    session_history.append(session_entry)
+    return session_history
+
+def save_conversation_to_file(session_history: list):
+    try:
+        history_dir = os.getenv('HISTORY_FOLDER_DIRECTORY')
+        if not history_dir:
+            raise ValueError("HISTORY_FOLDER_DIRECTORY environment variable is not set. Please add it to your .env file.")
+        
+        if not os.path.exists(history_dir):
+            os.makedirs(history_dir)
+        
+        timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+        filename = f"conversation_{timestamp}.txt"
+        filepath = os.path.join(history_dir, filename)
+        
+        with open(filepath, 'w', encoding='utf-8') as f:
+            f.write("Voyager T800 - Complete Conversation Session\n")
+            f.write("=" * 60 + "\n\n")
+            f.write(f"Session Date: {datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n")
+            f.write("=" * 60 + "\n\n")
+            
+            for i, entry in enumerate(session_history, 1):
+                f.write(f"ITINERARY #{i}\n")
+                f.write("-" * 40 + "\n")
+                f.write(f"Time: {entry['timestamp']}\n")
+                f.write(f"User Request: {entry['user_input']}\n\n")
+                f.write("Travel Preferences:\n")
+                for key, value in entry['preferences'].items():
+                    f.write(f"- {key}: {value}\n")
+                f.write("\n" + "-" * 40 + "\n")
+                f.write(entry['itinerary'])
+                f.write("\n\n" + "=" * 60 + "\n\n")
+        
+        return filepath
+        
+    except PermissionError as e:
+        return None
+    except OSError as e:
+        return None
+    except UnicodeEncodeError as e:
+        return None
+    except Exception as e:
+        return None
```

Comment:
Multiple except blocks all perform the same action (return None). Catch all relevant exceptions in a single tuple:
```
except (PermissionError, OSError, UnicodeEncodeError, Exception):
    return None
```